ID: 12543
Title: Invalid Input Error: executemany requires a list of parameter sets to be provided
Description:
### What happens?

In order to relieve memory presure I am using _generators_ (iterable) to provide the contents of what I want to insert into the database. This works well with the SQLite3 interface, sadly it produces an error in DuckDB.

### To Reproduce

```python
from typing import List, Generator

import duckdb

con = duckdb.connect()
con.execute("CREATE TABLE unittest_generator (a INTEGER);")

to_insert = [[1],[2],[3]]

con.executemany("INSERT into unittest_generator (a) VALUES (?)", to_insert)

def to_insert_from_generator(what: List) -> Generator:
    for x in what:
        yield x

con.executemany("INSERT into unittest_generator (a) VALUES (?)", to_insert_from_generator(to_insert))
```
```console
duckdb.duckdb.InvalidInputException: Invalid Input Error: executemany requires a list of parameter sets to be provided
```

Compare this to:

```python
from typing import List, Generator

import sqlite3

to_insert = [[1],[2],[3]]

def to_insert_from_generator(what: List) -> Generator:
    for x in what:
        yield x

con = sqlite3.connect(database=":memory:")
con.execute("CREATE TABLE unittest_generator (a INTEGER);")
con.executemany("INSERT into unittest_generator (a) VALUES (?)", to_insert)
con.executemany("INSERT into unittest_generator (a) VALUES (?)", to_insert_from_generator(to_insert))
```

### OS:

Linux

### DuckDB Version:

1.0.0

### DuckDB Client:

Python

### Full Name:

Stefan de Konink

### Affiliation:

Stichting OpenGeo

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have